packages/@sanity/core/src/util/checkRequiredDependencies.ts

```diff
@@ -0,0 +1,238 @@
+ * whether or not to add them to `package.json` and install them using yarn/npm
+ */
+export async function checkRequiredDependencies(context: CliCommandContext): Promise<void> {
+  const {workDir: studioPath, output} = context
+  const [
+    basePeerDependencies,
+    studioPackageManifest,
+    installedStyledComponentsVersion,
+  ] = await Promise.all([
+    await readBasePeerDependencies(studioPath),
+    await readPackageManifest(path.join(studioPath, 'package.json')),
+    await readModuleVersion(studioPath, 'styled-components'),
+  ])
+
+  // If the `@sanity/base` package.json does not have a `styled-components` peer dependency
+  // declared, the user is probably running an old version of `@sanity/base`. This is a bit
+  // of an indeterminate state, so we'll just have to accept it and assume things will work.
+  const wantedStyledComponentsVersionRange = basePeerDependencies['styled-components']
+  if (!wantedStyledComponentsVersionRange) {
+    return
+  }
+
+  // The studio _must_ now declare `styled-components` as a dependency. If it's not there,
+  // we'll want to automatically _add it_ to the manifest and tell the user to reinstall
+  // dependencies before running whatever command was being run
+  const declaredStyledComponentsVersion = studioPackageManifest.dependencies['styled-components']
+  if (!declaredStyledComponentsVersion) {
+    const deps = {'styled-components': wantedStyledComponentsVersionRange}
+    await installDependenciesWithPrompt(deps, context)
+    return
+  }
+
+  // The declared version should be semver-compatible with the version specified as a
+  // peer dependency in `@sanity/base`. If not, we should tell the user to change it.
+  //
+  // Exception: Ranges are hard to compare. `>=5.0.0 && <=5.3.2 || ^6`... Comparing this
+  // to anything is going to be challenging, so only compare "simple" ranges/versions
+  // (^x.x.x / ~x.x.x / x.x.x)
+  if (
+    isComparableRange(declaredStyledComponentsVersion) &&
+    !semver.satisfies(
+      normalizeVersion(declaredStyledComponentsVersion),
+      wantedStyledComponentsVersionRange
+    )
+  ) {
+    output.warn(oneline`
+      Declared version of styled-components (${declaredStyledComponentsVersion})
+      is not compatible with the version required by @sanity/base (${wantedStyledComponentsVersionRange}).
+      This might cause problems!
+    `)
+  }
+
+  // Ensure the studio has _installed_ a version of `styled-components`
+  if (!installedStyledComponentsVersion) {
+    throw new Error(oneline`
+      Declared dependency \`styled-components\` is not installed - run
+      \`npm install\` or \`yarn\` to install it before re-running this command.
+    `)
+  }
+
+  // The studio should have an _installed_ version of `styled-components`, and it should
+  // be semver compatible with the version specified in `@sanity/base` peer dependencies.
+  if (!semver.satisfies(installedStyledComponentsVersion, wantedStyledComponentsVersionRange)) {
```

Comment:
you may want to wrap `installedStyledComponentsVersion` in `semver.minVersion`

```suggestion
  if (!semver.satisfies(semver.minVersion(installedStyledComponentsVersion), wantedStyledComponentsVersionRange)) {
```

Reply:
ah nevermind. the version comes from the package itself so there is no carrot or anything. this isn't needed